### DISCUSSION TITLE
[wildfly] progamatically add Application and resources on demand

### DISCUSSION THREAD

**Discussion by guykatz** [resteasy/resteasy#3578] [Q&A] [answered]
Hi all; (i am using wildfly 23)
I have a situation where I cant use web.xml or a write ahead of time an implementation of Application with @ApplicationPath annotation.
my constraint is that I should programatically register an Application (with its Path) and its resources (with their Path) depending on some code logic...
also, I would like to use JAX-RS API exclusively if possible (but this is a second priority...)
r these things possible?
thanks

**Comment by jamezp** [comment]
There is no standard way to do this. This is how it would be done in a servlet container https://jakarta.ee/specifications/restful-ws/3.1/jakarta-restful-ws-spec-3.1.html#servlet.

What are you trying to achieve? Do you know before deployment your application path?

**Reply by guykatz** [reply]
thanks for replying @jamezp 
I am running inside a wildfly container inside an ear file with several wars, some of which already have JAX RS endpoints using Application and @Application path.
my need is to also add a new applicatiojn with a new application path dynamically depending if a condition inside an application is met or not...
so this is my goal...
generally speaking I would like to not know the path in advance but its not a deal breaker
hope this clarifies.
on a side note, will I be able to use SeBootstrap inside wildfly to achieve this maybe? thanks

**Reply by jamezp** [reply, marked as answer]
To my knowledge, without a bunch of internal code usages, there is no way to create an application on the fly. A deployment really needs to be processed by the deployment processors in WildFly.

No, you don't want to use `SeBootstrap` inside of WildFly. You'd be booting up two different servers and could run into issues. Ports already in-use would probably be a big one.

**Comment by guykatz** [comment]
is there a way to programatically tap into the container and "instruct" it to add a new application path, application and resources?
this w2ould also be good for my use case
BTW, what about RuntimeDelegate.createEndpoint? can this work in wildfly?

**Reply by jamezp** [reply]
Possibly, but you're getting into very specific WildFly use-cases. You'd probably have to develop a WildFly subsystem.

With RESTEasy you _might_ be able use a `ResteasyDeployment` to do this. However, I've not tested this and it might not work. It also requires RESTEasy internals which is something that is not technically supported in WildFly and could break in future releases.

Do you have an example/use-case? I'm wondering if there is another way to do it is why I ask.

**Comment by jamezp** [comment]
One of the reasons I ask what the end goal is, there is a way to dynamically register some things, just not applications https://jakarta.ee/specifications/restful-ws/3.1/jakarta-restful-ws-spec-3.1.html#dynamic_binding.

**Reply by guykatz** [reply]
thanks a lot. the perfect architecture for my use case needs everything to be constructed dynamically. it is a shame its not covered by the jaxrs spec.
however, with all the info I read and you provided I can probably do something that i can probably live with :)
thanks a lot./
